### ISSUE TITLE
Fixed header with reveal prop doesn't reshow itself on page navigation while loading plugin is active

### ISSUE THREAD

**Issue by kalvenschraut** [quasarframework/quasar#12994]
### What happened?

When scrolled down on a page with the fixed header hidden due to the reveal prop and clicking on a link that navigates to another page. If the page has a q loading plugin active then the header will not reappear. This is occuring do the fact the q loading plugin sets qScrollPrevented to true. If qScrollPrevented is true when the onPageScroll function is called in QLayout via the onScroll event then the $Layout.scroll is never updated which the QHeader watches.

### What did you expect to happen?

For the QHeader to not appear disappeared on the page navigated to.

### Reproduction URL

https://github.com/kalvenschraut/vitejs-vite-uswlyh/tree/master

### How to reproduce?

I had a lot of issues reproducing consistently on the stackblitz site but I have narrowed down what exactly the problem by debugging in my applications code. I feel like the scroll object needs to be updated regardless if scroll is prevented for cases like mine.

### Flavour

Vite Plugin (@quasar/vite-plugin)

### Areas

Components (quasar), Plugins (quasar)

### Platforms/Browsers

Chrome

### Quasar info output

```Shell
System:
    OS: Linux 5.10 Gentoo/Linux
    CPU: (4) x64 Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz
    Memory: 6.65 GB / 31.33 GB
    Container: Yes
    Shell: 5.1.8 - /bin/bash
  Binaries:
    Node: 16.10.0 - /usr/bin/node
    npm: 6.14.15 - /usr/bin/npm
    Watchman: 4.9.0 - /usr/bin/watchman
  Browsers:
    Chromium: 95.0.4638.69
    Firefox: 91.3.0esr
  npmPackages:
    @quasar/extras: ^1.13.4 => 1.13.4
    @quasar/quasar-ui-qcalendar: 4.0.0-beta.13 => 4.0.0-beta.13
    @quasar/vite-plugin: ^1.0.7 => 1.0.7
    @vitejs/plugin-vue: ^2.3.1 => 2.3.1
    quasar: 2.6.2 => 2.6.2
```

**Comment by kalvenschraut** [comment]
Stack blitz should be an allowed place for a repro.... exported it to github I Guess

**Comment by kalvenschraut** [comment]
My solution would be adding the data.position.top !== scroll.value.position to the if check OR just remove document.qScrollPRevented !== true completely, but I don't know what this all would affect. I can make a PR if needed

```ts
if (props.container === true || data.position.top !== scroll.value.position || document.qScrollPrevented !== true)
```

**Comment by freakzlike** [comment]
The issue still exists. I've created a workaround, which temporary resets the `reveal` prop on route change:
```vue
<template>
  <q-header :reveal="revealActive">
    <!-- ... -->
  </q-header>
</template>

<script setup lang="ts">
// ...
const revealActive = ref(true)
const hookCleanup = router.beforeEach(() => {
  revealActive.value = false
  nextTick(() => {
    revealActive.value = true
  })
})
onBeforeUnmount(() => {
  hookCleanup()
})
</script>
```

**Comment by rstoenescu** [comment]
Closing for lack of a reproduction after 2+ years. Comment with one and we'll reopen.

**Comment by kalvenschraut** [comment]
Lack of reproduction??? There is a stackblitz link which is I updated to github

**Comment by rstoenescu** [comment]
You're right, and apologies: the close reason was wrong. The repro link was there and your analysis was correct.

Confirmed root cause: while the Loading plugin holds the scroll lock, QLayout intentionally ignores the scroll-to-top that engaging the lock itself causes (that guard fixed #7012). When the route changes while the lock is held, releasing it skips the scroll restore, and since no further scroll event ever fires, QLayout keeps the stale pre-navigation scroll state; a `reveal` QHeader (and QFooter / QPageScroller) never recovers at the top of the new page.

About the suggested one-liner: letting updates through whenever the position differs would re-trigger #7012, since the lock's own scroll-to-top always differs from the stored position. The fix (a915d153d on dev) instead buffers the update suppressed during the lock and applies it when the lock releases without restoring the scroll position. Ships in v2.25.1.
